### PR TITLE
Update `send_payment_along_path` to take its args as struct

### PR DESCRIPTION
**Pull request by valentinewallace**
Cleans up and paves the way for additional arguments to be added more easily.

Edit: I thought this was helpful for route blinding, but it actually isn't. Still, I think it's a nice cleanup, so I'll keep it open.

### CODE REVIEW

**Review by vladimirfomene — COMMENTED** [review]
Code review ACK 80f904c5ebdd74f81eed4f853ed136ef5a751bbe. It definitely makes the code cleaner.

**Review by dunxen — APPROVED** [review]

File: lightning/src/ln/outbound_payment.rs [hunk 1]
```diff
@@ -473,6 +473,18 @@ impl RecipientOnionFields {
 	}
 }
 
+/// Arguments for [`super::channelmanager::ChannelManager::send_payment_along_path`].
+pub(super) struct SendAlongPathArgs<'a> {
```
Comment (henghonglee) [inline lightning/src/ln/outbound_payment.rs:477]: nit: Not so sure about the location of this struct. i'd think its supposed to be more co-located with where its used but i also think `channelmanager.rs` is also too large. 

what is the current understanding around `outbound_payment.rs`?  are we intending to move most/all outbound payment / routing / orchestration logic here?

**Review by henghonglee — COMMENTED** [review]
ACK. definitely makes the code cleaner. I do have a small question about `outbound_payment.rs`

**Review by alecchendev — APPROVED** [review]

File: lightning/src/ln/outbound_payment.rs (hunk 1 above)
Comment (valentinewallace) [reply lightning/src/ln/outbound_payment.rs:477]: The `outbound_payment` module came about so we could remove code from `channelmanager`. So, anything exclusive to outbound payments (such as this struct) could reasonably live there IMO. I believe we've moved most if not all of it already.

**Pull request merged**